pandas-stubs/core/indexes/base.pyi

```diff
@@ -88,12 +89,11 @@ class Index(IndexOpsMixin[S1]):
     @overload
     def __new__(  # type: ignore[misc]
         cls,
-        data: Iterable[float | np.float32 | np.float64],
+        data: Sequence[float | np.floating]
+        | IndexOpsMixin[float]
+        | npt.NDArray[np.floating],
```

Comment:
Maybe we should introduce a `np_ndarray_anyfloat` in `_typing` and use that here for consistency.